online-shop/src/components/ProductDetails.tsx

```diff
@@ -0,0 +1,82 @@
+import React from 'react'
+import '../productModeling/product.css';
+import { IProduct } from '../App';
+import PostData from '../products.json';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import { library, icon } from '@fortawesome/fontawesome-svg-core'
+import { faShoppingBasket,faStepBackward } from '@fortawesome/free-solid-svg-icons'
+import { Link } from 'react-router-dom';
+
+library.add(faShoppingBasket,faStepBackward)
+
+export interface IProps {
+    data?:IProduct;
+    match?:any;
+    appSetArray?:any;
```

Comment:
This should be typed as function: `appSetArray`: (products: IProduct[]) => void.

Also, you can find a better name for it. E.g. onAddProduct or onAddButtonClick. Usually callbacks which are linked to events start with `on`.

Also, you don't need to pass an array. This component should just be able to add one more product to the shoppin cart - it should be enough to pass just the product itself.